packages/web-ui/src/TextLink/TextLink.tsx

```diff
@@ -7,9 +7,20 @@ import { useBackground } from '../Box';
 export interface TextLinkProps
   extends React.AnchorHTMLAttributes<HTMLAnchorElement>,
     Pick<MuiLinkProps, 'children' | 'classes' | 'sx' | 'variant'> {
+  /**
+   * Sets textTransform: capitalize | lowercase | uppercase.
+   */
   textTransform?: MuiTypographyProps['textTransform'];
 }
 
+/**
+ * TextLink is for non button type links.
+ * It adds the data-heading attribute as true if variant is one of the following:
+ * 'displayHeading', 'h1', 'h2', 'h3', 'h4'.
+ * It adds the data-bg-color-brand attribute as true if nested in a Box component where
+ * background = isBrandBackground.
```

Comment:
```suggestion
```

These are internal concerns, consumers a) don't need to know this b) shouldn't expect to rely on this information not changing in the future.